File: fs18_CSharp_FullStack_Backend/BlossomAvenue.Core/Products/Product.cs
using System;
using System.Collections.Generic;
using BlossomAvenue.Core.ProductReviews;

namespace BlossomAvenue.Core.Products
{
    public class Product
    {
        public Guid ProductId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public ICollection<Image> Images { get; set; }
        public ICollection<Variation> Variations { get; set; }
        public ICollection<ProductCategory> ProductCategories { get; set; }

        public ICollection<ProductReview> ProductReviews { get; set; }
    }
}

File: fs18_CSharp_FullStack_Frontend/src/data/types/product.ts
export interface Variation {
    variationId: string;
    variationName: string;
    price: number;
    inventory: number;
    productId: string;
}

export interface Category {
    categoryId: string;
    categoryName: string;
    parentId: string | null;
}

export interface ReadProductReviewDto {
    reviewId: string;
    review: string | null;
    star: number | null;
    userName: string;
}

export interface GetProductByIdReadDto {
    productId: string;
    title: string;
    description: string;
    imageUrl: string | null;
    minPrice: number;
    inventory: number;
    avgStar: number;
    variation: Variation[];
    // categories: Category[];
    // productReviews: ReadProductReviewDto[];

}

export interface VariationInterface {
    variationName: string,
    price: number,
    inventory: number
}

export interface ImageInterface {
    imageUrl: string
}

export interface CategoryInterface {
    categoryId: string
}
export interface CreateProducDto {
    title: string,
    description: string,
    images: ImageInterface[],
    variations: VariationInterface[],
    productCategories: CategoryInterface[]

}

export interface UpdateVariationInterface extends VariationInterface {
    variationId: string,
    productId: string
}

export interface  UpdateImageInterface extends ImageInterface {
    imageId: string,
    productId: string
}

export interface UpdateCategoryInterface extends CategoryInterface {
    productCategoryId: string,
    productId: string
}
export interface UpdateProducDto {
    productId: string,
    title: string,
    description: string,
    images: UpdateImageInterface[],
    variations: UpdateVariationInterface[],
    productCategories: UpdateCategoryInterface[]
}